### PR TITLE
Adds unittest2

### PR DESCRIPTION
**Pull request by tgaspar**
Added [unittest2](https://pypi.org/project/unittest2/) package to `python.yml`:
> unittest2 is a backport of the new features added to the unittest testing framework in Python 2.7 and onwards.

Ubuntu
https://packages.ubuntu.com/disco/python-unittest2
Debian
https://packages.debian.org/stretch/python-unittest2
Fedora
https://apps.fedoraproject.org/packages/python-unittest2
Pip
https://pypi.org/project/unittest2/

### CODE REVIEW

**Comment by nuclearsandwich** [comment]
@tgaspar can you please add links to the Ubuntu, Debian, and Fedora repository web listings in addition to the pip one you've already linked as requested in the [contribution guide](https://github.com/ros/rosdistro/blob/master/CONTRIBUTING.md#rosdep-rules-contributions).

**Comment by tgaspar** [comment]
The repository web listing:
[Ubuntu](https://packages.ubuntu.com/disco/python-unittest2)
[Debian](https://packages.debian.org/stretch/python-unittest2)
[Fedora](https://apps.fedoraproject.org/packages/python-unittest2)

**Comment by tgaspar** [comment]
Updated my original comment for better clarity following the the recommendations from @nuclearsandwich. 

Now waiting for final review.

**Review by clalancette — APPROVED** [review]

**Review by cottsay — APPROVED** [review]

**Pull request merged**
